packages/data-stores/src/templates/use-templates.ts

```diff
@@ -0,0 +1,26 @@
+import { useQuery, UseQueryResult, QueryOptions } from '@tanstack/react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+import type { Template } from './types';
+
+interface Options extends QueryOptions< Template[] > {
+	enabled?: boolean;
+}
+
+const useTemplates = (
+	siteId: string | number,
+	queryOptions: Omit< Options, 'queryKey' > = {}
+): UseQueryResult< Template[] > => {
+	return useQuery< Template[] >( {
+		queryKey: [ siteId, 'templates' ],
+		queryFn: () =>
+			wpcomRequest( {
+				path: `/sites/${ siteId }/templates`,
```

Comment:
Note that we have to make changes to support forwarding templates endpoint to the atomic sites.

Reply:
Thanks for the reminder! I've added the endpoint to the API mapper via D133687-code.